### PR TITLE
Further NRT amends

### PR DESCRIPTION
**Pull request by AndyButland**
Resolves a few more NRT related issues that came to light in applying the same to Deploy:

- Deploy interfaces and base classes
- Template collection responses
- Content display variant collection
- `IMediaService.GetRootMedia` collection
- `IMemberService.GetMembersByMemberType` collection
- `ILocalizationService.GetRootDictionaryItems` and `GetDictionaryItemChildren` collections

### CODE REVIEW

File: src/Umbraco.Core/Models/Mapping/ContentVariantMapper.cs [hunk 1]
```diff
@@ -19,12 +19,12 @@ public ContentVariantMapper(ILocalizationService localizationService, ILocalized
...
         }
 
-        public IEnumerable<TVariant>? Map<TVariant>(IContent source, MapperContext context) where TVariant : ContentVariantDisplay
+        public IEnumerable<TVariant> Map<TVariant>(IContent source, MapperContext context) where TVariant : ContentVariantDisplay
         {
             var variesByCulture = source.ContentType.VariesByCulture();
             var variesBySegment = source.ContentType.VariesBySegment();
 
-            IList<TVariant>? variants = new List<TVariant>();
+            var variants = new List<TVariant>();
```
Comment (Zeegaan) [inline src/Umbraco.Core/Models/Mapping/ContentVariantMapper.cs:27]:
```suggestion
            List<TVariant> variants = new ();
```
Kinda nitpicky, but i like this new syntax a lot more 👍

File: src/Umbraco.Core/Models/Membership/MemberExportModel.cs [hunk 2]
```diff
@@ -10,10 +7,10 @@ public class MemberExportModel
         public string? Name { get; set; }
         public string? Username { get; set; }
         public string? Email { get; set; }
-        public List<string?>? Groups { get; set; }
+        public List<string> Groups { get; set; } = new List<string>();
```
Comment (Zeegaan) [inline src/Umbraco.Core/Models/Membership/MemberExportModel.cs:10]:
```suggestion
        public List<string> Groups { get; set; } = new ();
```
Same again 👍

File: src/Umbraco.Core/Models/Membership/MemberExportModel.cs [hunk 3]
```diff
@@ -10,10 +7,10 @@ public class MemberExportModel
...
         public string? ContentTypeAlias { get; set; }
         public DateTime CreateDate { get; set; }
         public DateTime UpdateDate { get; set; }
-        public List<MemberExportProperty>? Properties { get; set; }
+        public List<MemberExportProperty> Properties { get; set; } = new List<MemberExportProperty>();
```
Comment (Zeegaan) [inline src/Umbraco.Core/Models/Membership/MemberExportModel.cs:14]:
```suggestion
        public List<MemberExportProperty> Properties { get; set; } = new ();
```

File: src/Umbraco.Core/Services/MemberService.cs [hunk 4]
```diff
@@ -894,17 +894,17 @@ public IEnumerable<IMemberGroup> GetAllRoles()
...
         /// <returns>A list of member roles</returns>
-        public IEnumerable<string?>? GetAllRoles(int memberId)
+        public IEnumerable<string> GetAllRoles(int memberId)
         {
             using (ICoreScope scope = ScopeProvider.CreateCoreScope(autoComplete: true))
             {
                 scope.ReadLock(Constants.Locks.MemberTree);
                 var result = _memberGroupRepository.GetMemberGroupsForMember(memberId);
-                return result.Select(x => x.Name).Distinct();
+                return result.Where(x => x.Name is not null).Select(x => x.Name!).Distinct();
```
Comment (Zeegaan) [inline src/Umbraco.Core/Services/MemberService.cs:903]:
```suggestion
                return result..Select(x => x.Name).WhereNotNull().Distinct();
```
We have an extension method for this 👍

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Found some minor things 👓

File: src/Umbraco.Core/Models/Mapping/ContentVariantMapper.cs (hunk 1 above)
Comment (AndyButland) [reply src/Umbraco.Core/Models/Mapping/ContentVariantMapper.cs:27]: Time I got used to it I guesss!

File: src/Umbraco.Core/Services/MemberService.cs (hunk 4 above)
Comment (AndyButland) [reply src/Umbraco.Core/Services/MemberService.cs:903]: Useful, thanks.

**Review by Zeegaan — APPROVED** [review]
After the changes I am happy 👍 
Will merge once the last test has passed 🚀

**Pull request merged**